File: src/ReverseString.scala
trait ReverseString {
  def reverseString(s: String): String = {
    if (s == null || s.isEmpty) s
    else {
      var stack = List.empty[Char]
      for (ch <- s) stack = ch :: stack
      val sbf = new StringBuilder
      for (ch <- stack) sbf.append(ch)
      sbf.result()
    }
  }
}

object ReverseStringApp extends ReverseString with App {
   println(reverseString("abc"))
}

File: src/main/scala/mock/25102020/ReverseString.py
class Solution:
    def reverseString(self, s):
        if not s:
            return s
        i = 0
        j = len(s)-1
        while i < j:
            s[i],s[j] = s[j],s[i]
            i += 1
            j -= 1
        return s
